### ISSUE TITLE
spam doesn't work on linux - wrong javac location

### ISSUE THREAD

**Issue by Sciss** [scaled/scaled#11]
I'm getting this:

```
hhrutz@holiday:~ $ spam list
✓︎ Checking out Scaled package manager into /home/hhrutz/.scaled/Packages/pacman
Cloning into 'pacman'...
remote: Counting objects: 1068, done.
remote: Total 1068 (delta 0), reused 0 (delta 0), pack-reused 1068
Receiving objects: 100% (1068/1068), 176.90 KiB | 0 bytes/s, done.
Resolving deltas: 100% (484/484), done.
✓︎ Compiling Scaled package manager...
Exception in thread "main" java.io.IOException: Cannot run program "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/javac" (in directory "/home/hhrutz/.scaled/Packages/pacman"): error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at scaled.pacman.Bootstrap.exec(Bootstrap.java:289)
	at scaled.pacman.Bootstrap.buildPacman(Bootstrap.java:247)
	at scaled.pacman.Bootstrap.main(Bootstrap.java:70)
Caused by: java.io.IOException: error=2, No such file or directory
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 3 more
```

On Linux, you should find `javac` in the default PATH already. Here it's 

    /usr/bin/javac -> /etc/alternatives/javac -> /usr/lib/jvm/java-8-openjdk-amd64/bin/javac

**Comment by samskivert** [comment]
Pacman tries to enumerate all JDKs on the system and read metadata for them to determine, for example, what version of the JDK they are. In theory that allows it to select the right version of the JDK for projects that depend on Java 7 or Java 8 or whatever. In practice everything Scaled-related depends on Java 8 right now, so that's not the most important feature in the world, however I'm still reluctant to just rely on the javac in the user's path and assume that happens to be the right version of the JDK for our needs.

I fixed the JDK scanning code in Pacman to look in the standard place on Ubuntu (and maybe other Linuxes?), `/usr/lib/jvm`, but it's running into a problem which is that the openjdk package does not install a `release` file like the Oracle JDK packages do. This `release` file contains metadata on the JDK installation, for example:

```
JAVA_VERSION="1.8.0_144"
OS_NAME="Darwin"
OS_VERSION="11.2"
OS_ARCH="x86_64"
SOURCE=" .:57a5aae91b8b corba:0d5d2e29f93b deploy:9caf2c56add8 hotspot:db36f4d498b1 hotspot/make/closed:fceea951beb9 hotspot/src/closed:263c8c314db8 hotspot/test/closed:d6e0b3901dd8 install:2f28bfde8851 jaxp:330bd721dee5 jaxws:d2226ba553ca jdk:d2744852f3e6 jdk/make/closed:b74ada85df07 jdk/src/closed:4aa350c9c623 jdk/test/closed:87f0e789e983 langtools:816907853a15 nashorn:7de0a688b0d9 pubs:dbfc126a45a6 sponsors:21c32be63984"
BUILD_TYPE="commercial"
```

I guess I can modify the code to obtain release information in some other way if it sees what looks like a JAVA_HOME but which lacks a `release` file. I don't see any other file installed with the openjdk-8-jdk package that says anything about the version of the JDK installed by that package.

**Comment by samskivert** [comment]
OK, hopefully this is fixed with the latest commit to `pacman`.
